jsettlers.graphics/src/main/java/jsettlers/graphics/map/controls/original/panel/selection/BuildingSelectionContent.java

```diff
@@ -332,8 +332,15 @@ private BuildingBackgroundPanel createNormalBuildingContent(BuildingState state)
 			layout.priority.setPriority(supported, building.getPriority());
 		}
 
-		if (building.getBuildingType().getWorkRadius() <= 0) {
-			layout.background.removeChild(layout.workRadius);
+		if (building.getBuildingType() == EBuildingType.DOCKYARD) {
+			layout.background.removeChild(layout.buttonWorkRadius);
+		} else {
+			layout.background.removeChild(layout.buttonDockPosition);
```

Comment:
@michaelzangl : please do this for me, because I do not understand exactly what you want